shopfloor/actions/search.py

```diff
@@ -38,7 +38,10 @@ def product_from_scan(self, barcode):
         model = self.env["product.product"]
         if not barcode:
             return model.browse()
-        product = model.search([("barcode", "=", barcode)], limit=1)
+        product = model.search(
+            ["|", ("barcode", "=", barcode), ("barcode_ids.name", "=", barcode)],
```

Comment:
we can split the domain generation to a separated method to be able to override it from the other module